Error: tools/buildsystem/src/tasks/build/copy-assets.ts(20,17): error TS2345: Argument of type '{ overwrite: boolean; }' is not assignable to parameter of type 'DestOptions'.

### Category
- [ ] Enhancement
- [X] Bug
- [ ] Question
- [ ] Documentation gap/issue

### Version

Please specify what version of the library you are using: [1.1.0]

### Steps to Reproduce
- I followed this guide https://github.com/pnp/pnpjs/blob/dev/docs-src/getting-started-dev.md to get started with dev.
- I pulled the *master* branch
- I tried `gulp build` so I can have my custom build

### Expected / Desired Behavior / Question
- Js files compiled

### Observed Behavior
Getting the following error. Here is the full stack.

```
C:\p\pnpjs (master)
λ npm install -g gulp-cli
C:\Users\xx\AppData\Roaming\npm\gulp -> C:\Users\xx\AppData\Roaming\npm\node_modules\gulp-cli\bin\gulp.js
+ gulp-cli@2.0.1
updated 1 package in 104.167s

C:\p\pnpjs (master)
λ gulp build
[15:14:50] Using gulpfile C:\p\pnpjs\gulpfile.js
[15:14:50] Starting 'clean'...
[15:14:50]   Cleaning directories: ./build, ./dist, ./debugging, ./serve, ./testing.
[15:14:50] Starting 'lint'...
[15:14:52] Starting 'bootstrap-buildsystem'...
[15:14:53]   Cleaned directories: ./build, ./dist, ./debugging, ./serve, ./testing.
[15:14:53] Finished 'clean' after 3.32 s
[15:14:57] Finished 'lint' after 6.8 s
[15:14:57] 'bootstrap-buildsystem' errored after 5.65 s
[15:14:57] Error: tools/buildsystem/src/tasks/build/copy-assets.ts(20,17): error TS2345: Argument of type '{ overwrite: boolean; }' is not assignable to parameter of type 'DestOptions'.
  Object literal may only specify known properties, and 'overwrite' does not exist in type 'DestOptions'.
tools/buildsystem/src/tasks/build/copy-package-file.ts(23,17): error TS2345: Argument of type '{ overwrite: boolean; }' is not assignable to parameter of type 'DestOptions'.
  Object literal may only specify known properties, and 'overwrite' does not exist in type 'DestOptions'.
tools/buildsystem/src/tasks/build/replace-debug.ts(24,17): error TS2345: Argument of type '{ overwrite: boolean; }' is not assignable to parameter of type 'DestOptions'.
  Object literal may only specify known properties, and 'overwrite' does not exist in type 'DestOptions'.
tools/buildsystem/src/tasks/build/replace-sp-http-version.ts(30,17): error TS2345: Argument of type '{ overwrite: boolean; }' is not assignable to parameter of type 'DestOptions'.
  Object literal may only specify known properties, and 'overwrite' does not exist in type 'DestOptions'.
tools/buildsystem/src/tasks/package/banner.ts(43,17): error TS2345: Argument of type '{ cwd: string; overwrite: boolean; }' is not assignable to parameter of type 'DestOptions'.
  Object literal may only specify known properties, and 'overwrite' does not exist in type 'DestOptions'.
tools/buildsystem/src/tasks/package/copy-assets.ts(20,17): error TS2345: Argument of type '{ overwrite: boolean; }' is not assignable to parameter of type 'DestOptions'.
  Object literal may only specify known properties, and 'overwrite' does not exist in type 'DestOptions'.
tools/buildsystem/src/tasks/package/copy-src.ts(21,17): error TS2345: Argument of type '{ overwrite: boolean; }' is not assignable to parameter of type 'DestOptions'.
  Object literal may only specify known properties, and 'overwrite' does not exist in type 'DestOptions'.

    at formatError (C:\Users\xx\AppData\Roaming\npm\node_modules\gulp-cli\lib\versioned\^3.7.0\format-error.js:20:10)
    at Gulp.<anonymous> (C:\Users\xx\AppData\Roaming\npm\node_modules\gulp-cli\lib\versioned\^3.7.0\log\events.js:41:15)
    at emitOne (events.js:116:13)
    at Gulp.emit (events.js:211:7)
    at Gulp.Orchestrator._emitTaskDone (C:\p\pnpjs\node_modules\orchestrator\index.js:264:8)
    at C:\p\pnpjs\node_modules\orchestrator\index.js:275:23
    at finish (C:\p\pnpjs\node_modules\orchestrator\lib\runTask.js:21:8)
    at cb (C:\p\pnpjs\node_modules\orchestrator\lib\runTask.js:29:3)
    at exec (C:\p\pnpjs\tools\gulptasks\build.js:37:13)
    at ChildProcess.exithandler (child_process.js:282:5)
```

### Additional
- I am using windows  7 enterprise
- I have python 2.7.13 installed

having similar problem with macOS
$ node --version
v9.10.1
npm --version
6.1.0

$ gulp build
[23:40:14] Using gulpfile ~/development/pnpjs/gulpfile.js
[23:40:14] Starting 'clean'...
[23:40:14]   Cleaning directories: ./build, ./dist, ./debugging, ./serve, ./testing.
[23:40:14] Starting 'lint'...
[23:40:15] Starting 'bootstrap-buildsystem'...
[23:40:15] Finished 'bootstrap-buildsystem' after 14 ms
[23:40:15]   Cleaned directories: ./build, ./dist, ./debugging, ./serve, ./testing.
[23:40:15] Finished 'clean' after 1.2 s
[23:40:16] Finished 'lint' after 2.45 s
[23:40:16] Starting 'build'...
[23:40:16] 'build' errored after 398 μs
[23:40:16] Error: Cannot find module '/Users/tomi/development/pnpjs/build/tools/buildsystem'
    at Function.Module._resolveFilename (module.js:543:15)
    at Function.Module._load (module.js:470:25)
    at Module.require (module.js:593:17)
    at require (internal/module.js:11:18)
    at Gulp.gulp.task (/Users/tomi/development/pnpjs/tools/gulptasks/build.js:48:20)
    at module.exports (/Users/tomi/development/pnpjs/node_modules/orchestrator/lib/runTask.js:34:7)
    at Gulp.Orchestrator._runTask (/Users/tomi/development/pnpjs/node_modules/orchestrator/index.js:273:3)
    at Gulp.Orchestrator._runStep (/Users/tomi/development/pnpjs/node_modules/orchestrator/index.js:214:10)
    at /Users/tomi/development/pnpjs/node_modules/orchestrator/index.js:279:18
    at finish (/Users/tomi/development/pnpjs/node_modules/orchestrator/lib/runTask.js:21:8)

Looks like this [23:47:55] Error: Cannot find module '/Users/tomi/development/pnpjs/**build**/tools/buildsystem'
That **build** folder does not exist.

--- a/pnp-build.js
+++ b/pnp-build.js
@@ -15,8 +15,8 @@ const defaultBuildPipeline = [
  */
 const config = {
 
-    // root location, relative 
-    packageRoot: path.resolve(".\\packages\\"),
+    // root location, relative
+    packageRoot: path.resolve("./packages/"),
 
     // the list of packages to be built, in order
     // can be a string name or a plain object with additional settings
@@ -27,7 +27,7 @@ const config = {
      *      "assets": string[], // optional, default is config.assets
      *      "buildChain": (ctx) => Promise<void>[], // optional, default is config.buildChain
      * }
-     * 
+     *
      */
     packages: [
         "logging",
@@ -54,10 +54,10 @@ const config = {
 
     // relative to the package folder
     assets: [
-        "..\\..\\LICENSE",
-        "..\\readme.md",
+        "../../LICENSE",
+        "../readme.md",
         "rollup.*.config.js",
-        "**\\*.md"
+        "**/*.md"
     ],
 
     // the set of tasks run on each project during a build

--- a/pnp-debug.js
+++ b/pnp-debug.js
@@ -7,8 +7,8 @@ const tasks = require("./build/tools/buildsystem").Tasks.Build,
 */
 const config = {
 
-    // root location, relative 
-    packageRoot: path.resolve(".\\debug"),
+    // root location, relative
+    packageRoot: path.resolve("./debug"),
 
     // the list of packages to be built, in order
     // can be a string name or a plain object with additional settings

--- a/pnp-package.js
+++ b/pnp-package.js
@@ -21,10 +21,10 @@ const config = {
     /**
      * The directory to which packages will be written
      */
-    outDir: path.resolve(".\\dist\\packages\\"),
+    outDir: path.resolve("./dist/packages/"),
 
-    // root location, relative 
-    packageRoot: path.resolve(".\\build\\packages\\"),
+    // root location, relative
+    packageRoot: path.resolve("./build/packages/"),
 
     // the list of packages to be packaged, in order
     // can be a string name or a plain object with additional settings
@@ -35,7 +35,7 @@ const config = {
      *      "assets": string[], // optional, default is config.assets
      *      "buildChain": (ctx) => Promise<void>[], // optional, default is config.buildChain
      * }
-     * 
+     *
      */
     packages: [
         "logging",

--- a/pnp-publish.js
+++ b/pnp-publish.js
@@ -12,8 +12,8 @@ const defaultPublishPipeline = [
 */
 const config = {
 
-    // root location, relative 
-    packageRoot: path.resolve(".\\dist\\packages\\"),
+    // root location, relative
+    packageRoot: path.resolve("./dist/packages/"),
 
     // the list of packages to be built, in order
     // can be a string name or a plain object with additional settings
@@ -24,7 +24,7 @@ const config = {
      *      "assets": string[], // optional, default is config.assets
      *      "buildChain": (ctx) => Promise<void>[], // optional, default is config.buildChain
      * }
-     * 
+     *
      */
     packages: [
         "logging",

--- a/tools/buildsystem/src/tasks/build/build-project.ts
+++ b/tools/buildsystem/src/tasks/build/build-project.ts
@@ -5,11 +5,11 @@ const log = require("fancy-log");
 import { exec } from "child_process";
 import { BuildContext } from "./context";
 
-const tscPath = ".\\node_modules\\.bin\\tsc";
+const tscPath = "./node_modules/.bin/tsc";
 
 /**
  * Builds the project based on the supplied tsconfig.json file
- * 
+ *
  * @param ctx The build context
  */
 export function buildProject(ctx: BuildContext) {
@@ -31,7 +31,7 @@ export function buildProject(ctx: BuildContext) {
 
 /**
  * Builds the project based on the supplied tsconfig.json file, overriding the build to produce es5
- * 
+ *
  * @param ctx The build context
  */
 export function buildProjectES5(ctx: BuildContext) {

--- a/tools/buildsystem/src/tasks/build/build-project.ts
+++ b/tools/buildsystem/src/tasks/build/build-project.ts
@@ -5,7 +5,7 @@ const log = require("fancy-log");
 import { exec } from "child_process";
 import { BuildContext } from "./context";
 
-const tscPath = "./node_modules/.bin/tsc";
+const tscPath = path.join("./node_modules/.bin/tsc");
 
 /**
  * Builds the project based on the supplied tsconfig.json file

--- a/tools/buildsystem/src/tasks/package/package-project.ts
+++ b/tools/buildsystem/src/tasks/package/package-project.ts
@@ -3,7 +3,7 @@ import { PackageContext } from "./context";
 import { exec } from "child_process";
 const path = require("path");
 
-const rollupPath = "./node_modules/.bin/rollup";
+const rollupPath = path.join("./node_modules/.bin/rollup");
 
 /**
  * Builds the project based on the supplied tsconfig.json file

--- a/tools/buildsystem/src/tasks/package/package-project.ts
+++ b/tools/buildsystem/src/tasks/package/package-project.ts
@@ -3,11 +3,11 @@ import { PackageContext } from "./context";
 import { exec } from "child_process";
 const path = require("path");
 
-const rollupPath = ".\\node_modules\\.bin\\rollup";
+const rollupPath = "./node_modules/.bin/rollup";
 
 /**
  * Builds the project based on the supplied tsconfig.json file
- * 
+ *
  * @param ctx The build context
  */
 export function packageProject(ctx: PackageContext) {

--- a/tools/buildsystem/src/tasks/package/uglify.ts
+++ b/tools/buildsystem/src/tasks/package/uglify.ts
@@ -3,12 +3,12 @@ import { PackageContext } from "./context";
 import { exec } from "child_process";
 const path = require("path");
 
-const uglifyPath = path.resolve(".\\node_modules\\.bin\\uglifyjs");
+const uglifyPath = path.resolve("./node_modules/.bin/uglifyjs");
 
 /**
  * Minifies the files created in es5 format into the target dist folder
- * 
- * @param ctx The build context 
+ *
+ * @param ctx The build context
  */
 export function uglify(ctx: PackageContext) {
 

--- a/tools/gulptasks/build.js
+++ b/tools/gulptasks/build.js
@@ -14,7 +14,7 @@ const gulp = require("gulp"),
     fs = require("fs"),
     cmdLine = require("./args").processConfigCmdLine;
 
-const tscPath = "./node_modules/.bin/tsc";
+const tscPath = path.join("./node_modules/.bin/tsc");
 
 // give outselves a single reference to the projectRoot
 const projectRoot = path.resolve(__dirname, "../..");

--- a/tools/gulptasks/build.js
+++ b/tools/gulptasks/build.js
@@ -1,6 +1,6 @@
 //******************************************************************************
 //* build.js
-//* 
+//*
 //* Defines a custom gulp task for compiling TypeScript source code into
 //* js files.  It outputs the details as to what it generated to the console.
 //******************************************************************************
@@ -14,7 +14,7 @@ const gulp = require("gulp"),
     fs = require("fs"),
     cmdLine = require("./args").processConfigCmdLine;
 
-const tscPath = ".\\node_modules\\.bin\\tsc";
+const tscPath = "./node_modules/.bin/tsc";
 
 // give outselves a single reference to the projectRoot
 const projectRoot = path.resolve(__dirname, "../..");

